src/differential_value_iteration/experiments/evaluation_convergence.py

```diff
@@ -40,21 +56,45 @@ def run(environments: Sequence[structure.MarkovRewardProcess],
           if np.sum(np.abs(changes)) <= convergence_tolerance and i > 1:
             converged = True
             break
-        print(f'step_size:{step_size:.5f}\tConverged:{converged}\tafter {i} iterations')
+        print(
+          f'step_size:{step_size:.5f}\tConverged:{converged}\tafter {i} iterations\tFinal Changes:{changes}')
 
 
 def main(argv):
   del argv  # Stop linter from complaining about unused argv.
 
+  algorithm_constructors = []
+
+  # Create constructors that only depends on params common to all algorithms.
+  if FLAGS.dvi:
+    betas = np.geomspace(start=FLAGS.dvi_minimum_beta,
+                         stop=FLAGS.dvi_maximum_beta,
+                         num=FLAGS.dvi_num_betas,
+                         endpoint=True)
+    for beta in betas:
+      dvi_algorithm = functools.partial(dvi.Evaluation, beta=beta,
+                                        initial_r_bar=FLAGS.dvi_initial_rbar)
+      algorithm_constructors.append(dvi_algorithm)
+
+  if FLAGS.rvi:
+    rvi_algorithm = functools.partial(rvi.Evaluation,
+                                      reference_index=FLAGS.rvi_reference_index)
+    algorithm_constructors.append(rvi_algorithm)
+
+  if not algorithm_constructors:
+    raise ValueError('No algorithms scheduled to run.')
+
   # Generate stepsizes log-spaced minimum and maximum supplied.
   step_sizes = np.geomspace(
       start=FLAGS.minimum_step_size,
       stop=FLAGS.maximum_step_size,
       num=FLAGS.num_step_sizes,
       endpoint=True)
+
   environments = [micro.mrp1, micro.mrp2, micro.mrp3]
 
   run(environments=environments,
+      algorithm_constructors=algorithm_constructors,
       step_sizes=step_sizes,
```

Comment:
For the sake of generality, would it make sense to have different algorithm_constructors for different step sizes? That way, run() need not have a for loop for step sizes.

Reply:
This might indeed be cleaner!

I often write code like this iteratively, simplifying as necessary. I think we can do whatever we like here.

When we run at scale with bigger infrastructure, each experiment will likely be a separate program launched with a very different-looking tool.

Reply:
Sounds good!